### PR TITLE
Adds EdgeX MongoDB Auth Config

### PR DESCRIPTION
**Pull request by trcox**
Adds support for user/password credentials
Adds timeouts

Signed-off-by: Tyler_Cox <Tyler_Cox@dell.com>

### CODE REVIEW

File: cmd/client/main.go [hunk 1]
```diff
@@ -23,14 +25,26 @@ import (
 )
 
 const (
-	port        int    = 7070
-	defMongoURL string = "mongodb://0.0.0.0:27017"
+	port        int	   = 7070
+	defMongoURL string = "localhost"
+	defMongoUsername string = "core"
+	defMongoPassword string = "password"
+	defMongoDatabase string = "coredata"
```
Comment (drasko) [inline cmd/client/main.go]: Is DB "coredata"? I think it should be more "registration"-something (I did not examine all Java code in details yet).

**Comment by drasko** [comment]
@trcox thanks for the PR, it looks good.

One general question to address is how do we handle copyright notices. I am not fan of comment headers - they introduce unnecessary clutter in the code (I think some research for the text like [this one](https://opensource.com/law/14/2/copyright-statements-source-files) is needed). I think something like author list, as they do on Docker project, should be sufficient: https://github.com/moby/moby/blob/master/AUTHORS

Any opinions?

File: cmd/client/main.go (hunk 1 above)
Comment (trcox) [reply cmd/client/main.go]: Yes, export-client currently uses the (https://github.com/edgexfoundry/export-client/blob/139a43db4e4c9036dd202cd0ba5bd58513899c37/src/main/resources/application.properties#L59)[coredata DB]. I'll discuss with Jim next week, but I agree that the two services shouldn't share this db at first glance, likely to cause deployment problems/security implications.

But, this works with what we have today (tested with the mongo container).

Comment (drasko) [reply cmd/client/main.go]: @trcox I've just sent a mail to our ML regarding tight microservices coupling. We should use the opportunity of Go-rewrite to look at these architectural issues as well.

**Comment by drasko** [comment]
@trcox I am waiting for @mijicd to take a look at PR and gives a few comments/reccomendations before I merge it.

**Pull request merged**